packages/plugin-layout/src/Layout.ts

```diff
@@ -0,0 +1,111 @@
+import { JWPlugin, JWPluginConfig } from '../../core/src/JWPlugin';
+import {
+    LayoutEngine,
+    Component,
+    componentReference,
+    LayoutEngineConstructor,
+    LayoutEngineConfig,
+} from './LayoutEngine';
+import { VNode } from '../../core/src/VNodes/VNode';
+import { ZoneIdentifier } from './ZoneNode';
+
+let AutomaticPublicId = 0;
+
+interface LayoutMap extends Map<LayoutEngineConstructor, LayoutEngine> {
+    get<A extends LayoutEngineConstructor = LayoutEngineConstructor>(Engine: A): InstanceType<A>;
+}
+
+export class Layout<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T> {
+    readonly engines: LayoutMap = new Map();
+    readonly loaders = {
+        layoutEngines: this.loadLayoutEngines,
+        layoutComponents: this.loadLayoutComponents,
+    };
+    commands = {
+        'UI-show': {
+            title: 'Show a layout component',
+            handler: this.show.bind(this),
+        },
+        'UI-hide': {
+            title: 'Hide a layout component',
+            handler: this.hide.bind(this),
+        },
+    };
+
+    async start(): Promise<void> {
+        for (const layoutEngine of this.engines) {
+            await layoutEngine[1].start();
+        }
+    }
+    async stop(): Promise<void> {
+        for (const layoutEngine of this.engines) {
+            await layoutEngine[1].stop();
+        }
+    }
+    // to add a vNode in a zone (will be clone)
+    async add(vNode: VNode, zone: ZoneIdentifier): Promise<void> {
+        const promises = [];
+        for (const layoutEngine of this.engines) {
+            const promise = layoutEngine[1].add(vNode, zone);
+            promises.push(promise);
+        }
+        await Promise.all(promises);
+    }
+    // to remove a component (instance or clonse) from the zone
+    async remove(ref: componentReference): Promise<void> {
+        const promises = [];
+        for (const layoutEngine of this.engines) {
+            promises.push(layoutEngine[1].remove(ref));
+        }
+        await Promise.all(promises);
+    }
+    // to show component (instance or clonse) inside the zone
+    async show(params: { ref: componentReference }): Promise<void>;
+    async show(ref: componentReference): Promise<void>;
+    async show(params: { ref: componentReference } | componentReference): Promise<void> {
```

Comment:
This syntactic sugar is not useful.
This is not a pattern that we use for defining command.

The pattern used for all command would look like this:
```typescript
async show(params: UIShowParams): Promise<void> {}
```

Reply:
There are two syntaxes, I put the two definitions. I don't see the use of your UIShowParams

Reply:
Yes but why do we need two syntax.
For syntactic sugar ? That only add an unnecessary code style.
Let's prefer consistency.